examples/parameter-loading/infection_manager.rs

```diff
@@ -0,0 +1,99 @@
+use ixa::context::Context;
+
+use ixa::define_rng;
+use ixa::global_properties::ContextGlobalPropertiesExt;
+use ixa::random::ContextRandomExt;
+
+use rand_distr::Exp;
+
+use crate::people::ContextPeopleExt;
+use crate::people::InfectionStatus;
+use crate::people::InfectionStatusEvent;
+use crate::Parameters;
+
+define_rng!(InfectionRng);
+
+fn schedule_recovery(context: &mut Context, person_id: usize) {
+    let parameters = context.get_global_property_value(Parameters).clone();
+    let infection_duration = parameters.infection_duration;
+    let recovery_time = context.get_current_time()
+        + context.sample_distr(InfectionRng, Exp::new(1.0 / infection_duration).unwrap());
+    context.add_plan(recovery_time, move |context| {
+        context.set_person_status(person_id, InfectionStatus::R);
+    });
+}
+
+fn handle_infection_status_change(context: &mut Context, event: InfectionStatusEvent) {
+    if matches!(event.updated_status, InfectionStatus::I) {
+        schedule_recovery(context, event.person_id);
+    }
+}
+
+pub fn init(context: &mut Context) {
+    context.subscribe_to_event::<InfectionStatusEvent>(move |context, event| {
+        handle_infection_status_change(context, event);
+    });
+}
+
+#[cfg(test)]
+mod test {
+    use crate::people::ContextPeopleExt;
+    use crate::people::InfectionStatus;
+    use crate::people::InfectionStatusEvent;
+    use ixa::context::Context;
+    use ixa::define_data_plugin;
+    use ixa::global_properties::ContextGlobalPropertiesExt;
+    use ixa::random::ContextRandomExt;
+    use ixa::define_global_property;
+    use serde::{Deserialize, Serialize};
+    use super::*;
+    define_data_plugin!(RecoveryPlugin, usize, 0);
+    
+    #[derive(Serialize, Deserialize, Debug, Clone)]
+    pub struct ParametersValues {
+        pub population: usize,
+        pub max_time: f64,
+        pub seed: u64,
+        pub foi: f64,
+        pub infection_duration: f64,
+    }
+    define_global_property!(Parameters, ParametersValues);
+    
+    fn handle_recovery_event(context: &mut Context, event: InfectionStatusEvent) {
+        if matches!(event.updated_status, InfectionStatus::R) {
+            *context.get_data_container_mut(RecoveryPlugin) += 1;
+        }
+    }
+
+    #[test]
+    fn test_handle_infection_change() {
```

Comment:
I can't get this test to pass and not sure what I'm doing wrong. I am trying to set Parameters as a Global Property with the values specified in `parameter_values`. It works in the scope of the test, but in the function to `schedule_recovery` the global property shows as not defined. Does anyone have any idea of what I'm doing wrong? @k88hudson-cfa @jasonasher @ekr-cfa @ChiragKumar9 @bbbruce

Reply:
You are defining the `Parameters` struct in the `test` module (i.e. inside of `mod test`), so it's in a separate namespace and thus not visible to the function `schedule_recovery` defined outside.

Reply:
I figured it was something like that. That's an outdated version though. Still, I would like to know why it works now by calling `parameters_loader::ParametersValues`. 

More importantly, I would like to understand what's the best way to define those structs in the tests.